### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by sarvan0506** [boris-kz/CogAlg#74] [Q&A] [answered]
Hi, I have few questions regarding this project,

1. What is the first level of achievable goal/goals for this project? (for example; detecting Coco dataset objects)
2. What is the long-term vision? What level of AGI is planned to be achieved?
3. I see the code is mostly following the traditional CV approach(such as writing filters manually to extract patterns). Is there a plan for Deep-Learning based approach?

Thanks

**Comment by lasred** [comment, marked as answer]
Hi Sarvana

For #3, I can say that there is no current plan to add deep learning 

From the readme 

'Proposed algorithm is a comparison-first alternative to deep learning, neither statistical nor neuromorphic'

**Reply by sarvan0506** [reply]
Thank you, I got it.

**Reply by boris-kz** [reply]
Hi Sarvana
1: We don't have specific plans, it could be automatic image vectorization, drafted in comp_slice_ file, or may have to wait till we have 3D alg, for video object recognition. Roadmap is described in wiki: https://github.com/boris-kz/CogAlg/wiki
2: Superhuman :)
3: Manual filters are initializations, they are supposed to be updated by feedback from higher levels, when we have it. We don't plan to use ANN because this alg is supposed to do the same thing: general learning, but better. One of the two is a loser.

**Reply by sarvan0506** [reply]
Thank you Boris!
